extensions/sql-database-projects/src/tools/netcoreTool.ts

```diff
@@ -36,16 +40,27 @@ export class NetCoreTool {
 
 	private static _outputChannel: vscode.OutputChannel = vscode.window.createOutputChannel(projectsOutputChannel);
 
+	private netCoreInstallationPresent: boolean | undefined;
+
+	private osPlatform: string = os.platform();
+
 	public async findOrInstallNetCore(): Promise<boolean> {
```

Comment:
With this new state (found but insufficient version), I'm wondering if this logic should be refactored slightly to either return an enum as a result (something like `{notFound, foundGood, foundOldVersion}`) so that we don't have to keep track of extra state outside the scope of this function and what it returns.

Reply:
I did introduce a new enum, not exactly here, but does a similar job that you explained here.